File: crossecore-csharp/Java/Java/Java_/InstanceofExpressionBase.cs
/* CrossEcore is a cross-platform modeling framework that generates C#, TypeScript, 
 * JavaScript, Swift code from Ecore models with embedded OCL (http://www.crossecore.org/).
 * The original Eclipse Modeling Framework is available at https://www.eclipse.org/modeling/emf/.
 * 
 * contributor: Simon Schwichtenberg
 */
 
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using oclstdlib;
using Ecore;
namespace Java_{
	public class InstanceofExpressionBase 
	:ExpressionImpl, InstanceofExpression
	{
		private Expression _leftOperand;
		public virtual Expression leftOperand
		{
			get {
			
				return _leftOperand;
			}
			set {
				if (value != _leftOperand) {
					NotificationChain msgs = null;
					if (_leftOperand != null){
						msgs = ((InternalEObject)_leftOperand).eInverseRemove(this, EOPPOSITE_FEATURE_BASE - Java_PackageImpl.INSTANCEOFEXPRESSION_LEFTOPERAND, null, msgs);
					}
					if (value != null){
						msgs = ((InternalEObject)value).eInverseAdd(this, EOPPOSITE_FEATURE_BASE - Java_PackageImpl.INSTANCEOFEXPRESSION_LEFTOPERAND, null, msgs);
					}
					msgs = basicSetLeftOperand(value, msgs);
					if (msgs != null) {
						msgs.dispatch();
					}
				}
				else if (eNotificationRequired()){
					eNotify(new ENotificationImpl(this, NotificationImpl.SET,Java_PackageImpl.INSTANCEOFEXPRESSION_LEFTOPERAND , value, value));
				}
				}
		}
		private TypeAccess _rightOperand;
		public virtual TypeAccess rightOperand
		{
			get {
			
				return _rightOperand;
			}
			set {
				if (value != _rightOperand) {
					NotificationChain msgs = null;
					if (_rightOperand != null){
						msgs = ((InternalEObject)_rightOperand).eInverseRemove(this, EOPPOSITE_FEATURE_BASE - Java_PackageImpl.INSTANCEOFEXPRESSION_RIGHTOPERAND, null, msgs);
					}
					if (value != null){
						msgs = ((InternalEObject)value).eInverseAdd(this, EOPPOSITE_FEATURE_BASE - Java_PackageImpl.INSTANCEOFEXPRESSION_RIGHTOPERAND, null, msgs);
					}
					msgs = basicSetRightOperand(value, msgs);
					if (msgs != null) {
						msgs.dispatch();
					}
				}
				else if (eNotificationRequired()){
					eNotify(new ENotificationImpl(this, NotificationImpl.SET,Java_PackageImpl.INSTANCEOFEXPRESSION_RIGHTOPERAND , value, value));
				}
				}
		}
		
		protected override EClass eStaticClass() {
			return Java_PackageImpl.Literals.INSTANCEOFEXPRESSION;
		}
		
		
		public NotificationChain basicSetLeftOperand(Expression newobj, NotificationChain msgs) {
			var oldobj = _leftOperand;
			_leftOperand = newobj;
			if (eNotificationRequired()) {
				var notification = new ENotificationImpl(this, NotificationImpl.SET, Java_PackageImpl.INSTANCEOFEXPRESSION_LEFTOPERAND, oldobj, newobj);
				if (msgs == null){
					msgs = notification;
				}
				else{
					msgs.add(notification);
				}
			}
			return msgs;
		}
		public NotificationChain basicSetRightOperand(TypeAccess newobj, NotificationChain msgs) {
			var oldobj = _rightOperand;
			_rightOperand = newobj;
			if (eNotificationRequired()) {
				var notification = new ENotificationImpl(this, NotificationImpl.SET, Java_PackageImpl.INSTANCEOFEXPRESSION_RIGHTOPERAND, oldobj, newobj);
				if (msgs == null){
					msgs = notification;
				}
				else{
					msgs.add(notification);
				}
			}
			return msgs;
		}
		
		public override Object eGet(int featureID, bool resolve, bool coreType) {
			switch (featureID) {
				case Java_PackageImpl.INSTANCEOFEXPRESSION_COMMENTS:
					return comments;
				case Java_PackageImpl.INSTANCEOFEXPRESSION_ORIGINALCOMPILATIONUNIT:
					return originalCompilationUnit;
				case Java_PackageImpl.INSTANCEOFEXPRESSION_ORIGINALCLASSFILE:
					return originalClassFile;
				case Java_PackageImpl.INSTANCEOFEXPRESSION_RIGHTOPERAND:
					return rightOperand;
				case Java_PackageImpl.INSTANCEOFEXPRESSION_LEFTOPERAND:
					return leftOperand;
			}
			return base.eGet(featureID, resolve, coreType);
		}
		
		
		public override void eSet(int featureID, object newValue) {
			switch (featureID) {
				case Java_PackageImpl.INSTANCEOFEXPRESSION_COMMENTS:
					comments.Clear();
					comments.AddRange(((List<EObject>)newValue)?.Cast<Comment>());
					return;
				case Java_PackageImpl.INSTANCEOFEXPRESSION_ORIGINALCOMPILATIONUNIT:
					originalCompilationUnit = (CompilationUnit) newValue;
					return;
				case Java_PackageImpl.INSTANCEOFEXPRESSION_ORIGINALCLASSFILE:
					originalClassFile = (ClassFile) newValue;
					return;
				case Java_PackageImpl.INSTANCEOFEXPRESSION_RIGHTOPERAND:
					rightOperand = (TypeAccess) newValue;
					return;
				case Java_PackageImpl.INSTANCEOFEXPRESSION_LEFTOPERAND:
					leftOperand = (Expression) newValue;
					return;
			}
			base.eSet(featureID, newValue);
		}
		
		
	}
}

File: crossecore-typescript/java_/InstanceofExpressionBase.ts
import {EClass} from "ecore/EClass";
import {Set} from "ecore/Set";
import {NotificationChain} from "ecore/NotificationChain";
import {ENotificationImpl} from "ecore/ENotificationImpl";
import {NotificationImpl} from "ecore/NotificationImpl";
import {InternalEObject} from "ecore/InternalEObject";
import {BasicEObjectImpl} from "ecore/BasicEObjectImpl";
import {Java_PackageLiterals} from "./Java_PackageLiterals";
import {ExpressionImpl} from "./ExpressionImpl";
import {Expression} from "./Expression";
import {InstanceofExpression} from "./InstanceofExpression";
import {TypeAccess} from "./TypeAccess";
		
			export class InstanceofExpressionBase
			extends ExpressionImpl
			implements InstanceofExpression
			{
				private _rightOperand:TypeAccess = null;
				get rightOperand():TypeAccess{
				
					return this._rightOperand;
				}
				set rightOperand(value:TypeAccess) {
					if (value != this._rightOperand) {
						let msgs:NotificationChain = null;
						if (this._rightOperand != null){
							msgs = (this._rightOperand).eInverseRemove(this, BasicEObjectImpl.EOPPOSITE_FEATURE_BASE - Java_PackageLiterals.INSTANCEOFEXPRESSION_RIGHTOPERAND, /*null*/ null , msgs);
						}
						if (value != null){
							msgs = value.eInverseAdd(this, BasicEObjectImpl.EOPPOSITE_FEATURE_BASE - Java_PackageLiterals.INSTANCEOFEXPRESSION_RIGHTOPERAND, /*null*/ null, msgs);
						}
						msgs = this.basicSetRightOperand(value, msgs);
						if (msgs != null) {
							msgs.dispatch();
						}
					}
					else if (this.eNotificationRequired()){
						this.eNotify(new ENotificationImpl(this, NotificationImpl.SET,Java_PackageLiterals.INSTANCEOFEXPRESSION_RIGHTOPERAND , value, value));
					}
				}
				private _leftOperand:Expression = null;
				get leftOperand():Expression{
				
					return this._leftOperand;
				}
				set leftOperand(value:Expression) {
					if (value != this._leftOperand) {
						let msgs:NotificationChain = null;
						if (this._leftOperand != null){
							msgs = (this._leftOperand).eInverseRemove(this, BasicEObjectImpl.EOPPOSITE_FEATURE_BASE - Java_PackageLiterals.INSTANCEOFEXPRESSION_LEFTOPERAND, /*null*/ null , msgs);
						}
						if (value != null){
							msgs = value.eInverseAdd(this, BasicEObjectImpl.EOPPOSITE_FEATURE_BASE - Java_PackageLiterals.INSTANCEOFEXPRESSION_LEFTOPERAND, /*null*/ null, msgs);
						}
						msgs = this.basicSetLeftOperand(value, msgs);
						if (msgs != null) {
							msgs.dispatch();
						}
					}
					else if (this.eNotificationRequired()){
						this.eNotify(new ENotificationImpl(this, NotificationImpl.SET,Java_PackageLiterals.INSTANCEOFEXPRESSION_LEFTOPERAND , value, value));
					}
				}

			
				public static eStaticClass:EClass;
				
				protected eStaticClass():EClass{
					
					return InstanceofExpressionBase.eStaticClass;
				}
			
			
				public basicSetLeftOperand(newobj:Expression, msgs:NotificationChain):NotificationChain {
					let oldobj = this._leftOperand;
					this._leftOperand = newobj;
					if (this.eNotificationRequired()) {
						let notification = new ENotificationImpl(this, NotificationImpl.SET, Java_PackageLiterals.INSTANCEOFEXPRESSION_LEFTOPERAND, oldobj, newobj);
						if (msgs == null){
							msgs = notification;
						}
						else{
							msgs.add(notification);
						}
					}
					return msgs;
				}
				public basicSetRightOperand(newobj:TypeAccess, msgs:NotificationChain):NotificationChain {
					let oldobj = this._rightOperand;
					this._rightOperand = newobj;
					if (this.eNotificationRequired()) {
						let notification = new ENotificationImpl(this, NotificationImpl.SET, Java_PackageLiterals.INSTANCEOFEXPRESSION_RIGHTOPERAND, oldobj, newobj);
						if (msgs == null){
							msgs = notification;
						}
						else{
							msgs.add(notification);
						}
					}
					return msgs;
				}
				
			
				public eGet_number_boolean_boolean(featureID:number, resolve:boolean, coreType:boolean):any{
					switch (featureID) {
						case Java_PackageLiterals.INSTANCEOFEXPRESSION_COMMENTS:
							return this.comments;
						case Java_PackageLiterals.INSTANCEOFEXPRESSION_ORIGINALCOMPILATIONUNIT:
							return this.originalCompilationUnit;
						case Java_PackageLiterals.INSTANCEOFEXPRESSION_ORIGINALCLASSFILE:
							return this.originalClassFile;
						case Java_PackageLiterals.INSTANCEOFEXPRESSION_RIGHTOPERAND:
							return this.rightOperand;
						case Java_PackageLiterals.INSTANCEOFEXPRESSION_LEFTOPERAND:
							return this.leftOperand;
					}
					//return this.eGetFromExpression(featureID, resolve, coreType);
					return super.eGet(featureID, resolve, coreType);
				}
				
				public eSet_number_any(featureID:number, newValue:any):void {
					switch (featureID) {
						case Java_PackageLiterals.INSTANCEOFEXPRESSION_RIGHTOPERAND:
							this.rightOperand = <TypeAccess> newValue;
							return;
						case Java_PackageLiterals.INSTANCEOFEXPRESSION_LEFTOPERAND:
							this.leftOperand = <Expression> newValue;
							return;
					}
					super.eSet_number_any(featureID, newValue);
				}

				
			}
			
